cpp/examples/basic.cpp

```diff
@@ -202,25 +202,27 @@ int main(int argc, char** argv)
 
   // Schedule small wireup messages to let UCX identify capabilities between endpoints
   requests.push_back(listener_ctx->getEndpoint()->tagSend(
-    sendWireupBuffer.data(), sendWireupBuffer.size() * sizeof(int), 0));
-  requests.push_back(
-    endpoint->tagRecv(recvWireupBuffer.data(), sendWireupBuffer.size() * sizeof(int), 0));
+    sendWireupBuffer.data(), sendWireupBuffer.size() * sizeof(int), ucxx::Tag{0}));
+  requests.push_back(endpoint->tagRecv(recvWireupBuffer.data(),
+                                       sendWireupBuffer.size() * sizeof(int),
+                                       ucxx::Tag{0},
+                                       ucxx::TagMaskFull));
   ::waitRequests(progress_mode, worker, requests);
   requests.clear();
 
   // Schedule send and recv messages on different tags and different ordering
   requests.push_back(listener_ctx->getEndpoint()->tagSend(
-    sendBuffers[0].data(), sendBuffers[0].size() * sizeof(int), 0));
+    sendBuffers[0].data(), sendBuffers[0].size() * sizeof(int), ucxx::Tag{0}));
   requests.push_back(listener_ctx->getEndpoint()->tagRecv(
-    recvBuffers[1].data(), recvBuffers[1].size() * sizeof(int), 1));
+    recvBuffers[1].data(), recvBuffers[1].size() * sizeof(int), ucxx::Tag{1}, ucxx::TagMaskFull));
```

Comment:
I'd prefer to leave it as is for now. Currently the only default parameters are those that are beyond the request's type, e.g. `enablePythonFuture` and`callbackFunction`/`callbackData`. I think we may want to revisit this API later and think how it would be best to order arguments and which ones should have defaults.